### ISSUE TITLE
Markdown render broken when safe mode enabled

### ISSUE THREAD

**Issue by afbora** [k-next/kirby#4598]
## To reproduce

1. Set `'markdown' => ['safe' => true]` from config file
2. Add markdown block
3. Add link with kirbytag into markdown block: `(link: https://duckduckgo.com text: Duck Duck Go)`
4. Go to front page and see issue

## Your setup

**Kirby Version**  

Kirby 3.7.4

**Additional context**  

Reversing the processing order of `kirbytags` and `markdown` seems to work fine.

https://github.com/getkirby/kirby/blob/3.7.4-rc.1/src/Cms/App.php#L915-L916

**Comment by lukasbestle** [comment]
I don't get an issue with the Markdown link syntax, but with `(link: https://duckduckgo.com text: Duck Duck Go)`

## Summary

Because we first apply KirbyTags, the Markdown parser already gets HTML code and will escape that in the safe mode. If we reverse it, the Markdown parser gets text and passes it along unchanged for the KirbyTags parser.

We need to check if reversing breaks anything else (e.g. can there be KirbyTags that would break when Markdown syntax is applied to them before the KirbyTags parser is run?).

Alternative, more thorough but more complex solution: First replace each KirbyTag with a random placeholder string, then run the Markdown parser and finally replace the interpreted KirbyTags back into the placeholders.

**Comment by afbora** [comment]
> I don't get an issue with the Markdown link syntax, but with (link: https://duckduckgo.com text: Duck Duck Go)

My bad. You're right. I've updated the description.

**Comment by distantnative** [comment]
Looking at the markdown syntax, the only parts that could potentially be in conflict with Kirbytags would be links and images:
```
[This link](http://example.net/) 
[Alt text](/path/to/img.jpg)
```

**Comment by lukasbestle** [comment]
No, it's really just KirbyTags because of the processing order. Parsedown safe mode can handle the Markdown syntax fine.

**Comment by distantnative** [comment]
You didn't understand me right. I meant that these markdown syntaxes are the only ones that Kirby tags could potentially be mistaken as. Still no direct match, but these are the two we need to test it in no case a Kirby tag gets mistaken for a link or image markdown.

**Comment by lukasbestle** [comment]
Hm, still not sure I understand you right.

What happens is that the KirbyTags generate HTML code. And the HTML code is escaped by Parsedown safe mode. There is no need for them to be mistaken for Markdown syntax.

**Comment by distantnative** [comment]
But we are talking about reversing this, first running the markdown parade. So we need to make sure that no Kirbytag gets mistaken as markdown and parsed differently than currently. And all I'm saying is that if at all, kirbytags could be mistakenly parsed as link or image first.

**Comment by lukasbestle** [comment]
Ah, now I understand what you mean.

I think apart from actual Markdown syntax, there will also be the issue that Parsedown will treat a KirbyTag that is alone in its line as a paragraph and wrap it with `<p>`, even if the KirbyTag outputs a block-level element.

I have the feeling the only proper solution can be to integrate KirbyTags deeply into Parsedown. But since we want to migrate away from Parsedown, that won't be viable.

**Comment by distantnative** [comment]
I tried to tackle this again, but no good solution
- Reversing the order alone creates issues: markdown will auto-link and we cannot be sure it will leave KirbyTags untouched
- Replacing KirbyTags first with a placeholder, then running markdown, then replacing placeholder with Kirbytags faces the problem that placeholders get wrapped in `<p>`

I think for the time being one has to stay away from markdown safe mode when using `->kirbytext()`.

If we switch to another markdown parser and build KirbyText on top of that we might have a change. But until then probably no good one.

**Comment by lukasbestle** [comment]
Note for later reference: Once this is fixed, we should make `markdown.safe = true` the default.
